lib/ansible/utils/plugin_docs.py

```diff
@@ -170,33 +174,35 @@ def add_fragments(doc, filename, fragment_loader, is_module=False):
         real_collection_name = '.'.join(real_fragment_name.split('.')[0:2]) if '.' in real_fragment_name else ''
         add_collection_to_versions_and_dates(fragment, real_collection_name, is_module=is_module)
 
-        if 'notes' in fragment:
-            notes = fragment.pop('notes')
-            if notes:
-                if 'notes' not in doc:
-                    doc['notes'] = []
-                doc['notes'].extend(notes)
-
-        if 'seealso' in fragment:
-            seealso = fragment.pop('seealso')
-            if seealso:
-                if 'seealso' not in doc:
-                    doc['seealso'] = []
-                doc['seealso'].extend(seealso)
-
-        if 'options' not in fragment and 'attributes' not in fragment:
-            raise Exception("missing options or attributes in fragment (%s), possibly misformatted?: %s" % (fragment_name, filename))
-
-        # ensure options themselves are directly merged
-        for doc_key in ['options', 'attributes']:
-            if doc_key in fragment:
-                if doc_key in doc:
-                    try:
-                        merge_fragment(doc[doc_key], fragment.pop(doc_key))
-                    except Exception as e:
-                        raise AnsibleError("%s %s (%s) of unknown type: %s" % (to_native(e), doc_key, fragment_name, filename))
-                else:
-                    doc[doc_key] = fragment.pop(doc_key)
+        if section == 'DOCUMENTATION':
+            # notes, seealso, options and attributes entries are specificly merged, but only occur in documentation section
+            if 'notes' in fragment:
+                notes = fragment.pop('notes')
+                if notes:
+                    if 'notes' not in doc:
+                        doc['notes'] = []
+                    doc['notes'].extend(notes)
+
+            if 'seealso' in fragment:
+                seealso = fragment.pop('seealso')
+                if seealso:
+                    if 'seealso' not in doc:
+                        doc['seealso'] = []
+                    doc['seealso'].extend(seealso)
```

Comment:
While you're touching these lines, how about factoring out common code, similarly as `options` and `attributes` are handled together further below:
```suggestion
            for doc_key in ['notes', 'seealso']:
                if doc_key in fragment:
                    entries = fragment.pop(doc_key)
                    if entries:
                        if doc_key not in doc:
                            doc[doc_key] = []
                        doc[doc_key].extend(entries)
```